### DISCUSSION TITLE
自定义主页：OpenBMCLAPI 仪表盘 Lite

### DISCUSSION THREAD

**Discussion by Silverteal** [Meloong-Git/PCL#4547] [自定义主页]
效果展示：

![新版](https://github.com/user-attachments/assets/5070b676-855f-4832-9eb1-0c704d5140e6)

地址：`https://pcl-bmcl.milu.ink/`

**Comment by ZeroWolf233** [comment]
这个可以有
实时查看下载负载和压力

**Comment by Mxmilu666** [comment]
这个可以有

**Reply by Mxmilu666** [reply]
我正在尝试此服务迁移到上海腾讯云的云函数中，希望表现会好一点...

**Reply by Silverteal** [reply]
> 我正在尝试此服务迁移到上海腾讯云的云函数中，希望表现会好一点...

有两个问题：
1. 境内的 Serverless 绑定自定义域名需要备案
2. 由于 API 网关功能下线了，好像根本没法绑定自定义域名，只能用它默认的一个版本一个链接的函数 URL，疑似不是很灵活
3. 以上两个问题可以通过另一个服务器手写 301/302/307/308 解决

**Reply by Mxmilu666** [reply]
> > 我正在尝试此服务迁移到上海腾讯云的云函数中，希望表现会好一点...
> 
> 有两个问题：
> 1. 境内的 Serverless 绑定自定义域名需要备案
> 2. 由于 API 网关功能下线了，好像根本没法绑定自定义域名，只能用它默认的一个版本一个链接的函数 URL，疑似不是很灵活
> 3. 以上两个问题可以通过另一个服务器手写 301/302/307/308 解决

我有备案域名，API 网关有另外的替代品

**Comment by SALTWOOD** [comment]
如果要投稿可以放到 #2528 去，要是不想投稿当我没说

**Reply by Silverteal** [reply]
我这里似乎不满足中国大陆快速响应的条件，你们可以测测看

**Reply by SALTWOOD** [reply]
要不用 93@Home 分布式分发（？）

**Reply by ZeroWolf233** [reply]
不如直接上oba的分布式发布
93@home常年就我俩节点和主控在线

**Reply by ZeroWolf233** [reply]
给93打钱，然后让他承载，在他的爱发电有着项目好像

**Reply by wuliaodexiaoluo** [reply]
> 给93打钱，然后让他承载，在他的爱发电有着项目好像

非常“便宜”（至少对于正经学生来说，这是一笔不小的负担）
~虽然开源可免费，但不排除 93 哥认为这种文件不符合仅限 MC 的要求（虽然不太可能）~

**Reply by SALTWOOD** [reply]
> 不如直接上oba的分布式发布 93@home常年就我俩节点和主控在线

云望的也在

**Reply by WorldHim** [reply]
> 要不用 93@Home 分布式分发（？）

可以考虑（？

**Comment by LTCatt** [comment]
所以加进预设的链接是用 `https://oba-bd.mcmirror.top/` 吗？

**Reply by ZeroWolf233** [reply]
我们正在寻找国内的服务器来作为主页的服务器，可以稍微等一下我们部署国内的服务器，那样速度会更快，目前最快的地址是pcl-bmcl.milulu.xyz

**Reply by Silverteal** [reply]
我这边这个部署的地址就是 `https://oba-bd.mcmirror.top/`，他们在搞上海云函数

**Reply by Mxmilu666** [reply]
``https://pcl-bmcl.milulu.xyz/``
已经部署上海腾讯云函数
由于腾讯云的一些原因目前采用的策略是 Cloudflare 302 跳转到腾讯云函数，速度理论秒开？（

**Reply by ZeroWolf233** [reply]
目前`https://pcl-bmcl.milulu.xyz/`的实际测速结果为在中国大陆境内可以在1-2s内打开，速度还行

**Reply by ZeroWolf233** [reply]
`https://oba-bd.mcmirror.top/`可以在0-1s内打开，速度更快

**Reply by SALTWOOD** [reply]
可以先不加，等等看

**Reply by Mxmilu666** [reply]
> `https://oba-bd.mcmirror.top/`可以在0-1s内打开，速度更快

`https://oba-bd.mcmirror.top/` 有缓存，无法获取实时数据，`https://pcl-bmcl.milulu.xyz/` 为实时获取，需要请求 API，所以速度会慢一点

**Reply by Silverteal** [reply]
然而 PCL 的默认逻辑是每次启动器启动时获取一次然后除非手动刷新否则不动，如果用户非要瞎 jb 刷新我的建议是直接去看 bd……不然几分钟的缓存刷新间隔应该不算太长。

然后现在这个云函数在 Cloudflare 层可以把 /version 直接拦掉，没用且耗时耗流，参考`https://ts-oba-bd.mcmirror.top/`，可能能快个几十几百毫秒？

**Reply by LTCatt** [reply]
你们可以讨论一下最后 PCL 用哪个，然后我下个版本就加进去 =。=

**Reply by Mxmilu666** [reply]
> 然而 PCL 的默认逻辑是每次启动器启动时获取一次然后除非手动刷新否则不动，如果用户非要瞎 jb 刷新我的建议是直接去看 bd……不然几分钟的缓存刷新间隔应该不算太长。
> 
> 然后现在这个云函数在 Cloudflare 层可以把 /version 直接拦掉，没用且耗时耗流，参考`https://ts-oba-bd.mcmirror.top/`，可能能快个几十几百毫秒？

好的，已经采取建议 /version 直接在 Cloudflare 层返回空内容

**Reply by Silverteal** [reply]
> 你们可以讨论一下最后 PCL 用哪个，然后我下个版本就加进去 =。=

用 https://pcl-bmcl.milulu.xyz/ 那个吧

**Comment by LTCatt** [comment]
“查看赞助商信息” 按钮是否应该跳转到 https://bd.bangbang93.com/pages/rank/clusters ，而非特定的一个赞助商？

**Reply by Mxmilu666** [reply]
> “查看赞助商信息” 按钮是否应该跳转到 https://bd.bangbang93.com/pages/rank/clusters ，而非特定的一个赞助商？

目前使用的是 ``https://bmclapi2.bangbang93.com/openbmclapi/sponsor`` 这个接口，如果跳转到整个排行榜的话也不是不行 (?

**Reply by weyeahh** [reply]
查看赞助商可以用 https://bmclapi2.bangbang93.com/openbmclapi/sponsor 这个接口随机获取一个放上去吧 跳转整个排行榜？emmm 不如这个接口
![BHR0%VH@DLLH5W353M@8TN1](https://github.com/user-attachments/assets/31095fd1-bb0a-41df-b01b-88d4778dca1e)

**Reply by Silverteal** [reply]
这个赞助商信息是通过后端调用 BMCLAPI 的单赞助商 API 然后返回到前端的，此接口设定上在一定时间内不会改变，所以看起来可能像是只链接到了一个赞助商，实际上是会动态改变的。

之前考虑过直接渲染赞助商提供内容而不是链接，后来考虑到排版等因素就没有实现。完整榜单在弹出的网页上左边点一下就能看到，也不是那么麻烦。

此外有些赞助商就是有很多节点，刷到的概率比较高。

**Reply by Silverteal** [reply]
> 查看赞助商可以用 https://bmclapi2.bangbang93.com/openbmclapi/sponsor 这个接口随机获取一个放上去吧 跳转整个排行榜？emmm 不如这个接口
> ![BHR0%VH@DLLH5W353M@8TN1](https://private-user-images.githubusercontent.com/121325752/363370502-31095fd1-bb0a-41df-b01b-88d4778dca1e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwOTI0ODcsIm5iZiI6MTcyNTA5MjE4NywicGF0aCI6Ii8xMjEzMjU3NTIvMzYzMzcwNTAyLTMxMDk1ZmQxLWJiMGEtNDFkZi1iMDFiLTg4ZDQ3NzhkY2ExZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMVQwODE2MjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYzYwY2VhMzJmMjAxNWY4MjhiODg3MzgyN2UyNzQ1ZDc4ODYxNDUzMjc3ZTJhM2RkOGQyMjEwZDljNTczMDFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1AMBkJHDAmQXUrQXWjY2WTpa032NRPFEZGtlzgSlgr4)

你说的这个就是当前行为

**Reply by LTCatt** [reply]
行（

**Comment by Mxmilu666** [comment]
关于主页加载过慢的问题我们正在想办法解决(

**Reply by Mxmilu666** [reply]
(顺带说一声开学了回复不及时各位见谅w

**Reply by weyeahh** [reply]
套了CF free 肯定加载慢 而且是实时请求api获得当前数据

**Comment by Silverteal** [comment]
提醒：`mcmirror.top`域名的主页链接即将失效，请切换到预设

**Comment by Mxmilu666** [comment]
推荐使用新主页链接
`https://pcl-bmcl.milu.ink/` 
已经全部切到中国内地了，访问速度应该不会差了
前一个链接会提供服务至 `2025-06-01` 为止

**Reply by Mxmilu666** [reply]
@LTCatt 麻烦帮忙换一下链接（

**Reply by Joker2184** [reply]
得下个版本了（）
昨天说正好赶上更新abb

**Reply by LTCatt** [reply]
下个版本.jpg
